### PR TITLE
[#3091] Cancel large iput -f test (master)

### PR DESCRIPTION
**Pull request by alanking**
Adds a test similar to test_cancel_large_iput but for a forced overwrite
of a data object. The test especially ensures that the checksum is
updated and is not equal to the old checksum which would be incorrect
given the partial overwrite of the data.

---

As this only adds a test, I only ran the one test and it passes. The old test which was renamed passes as well.

### CODE REVIEW

File: scripts/irods/test/test_icommands_file_operations.py [hunk 1]
```diff
@@ -683,6 +683,70 @@ def test_cancel_large_iput(self):
...
+            if not os.path.exists(large_local_file):
+                lib.make_file(large_local_file, file_size)
+
+            iput_cmd = 'iput -f -K {0} {1}'.format(large_local_file, logical_path)
+            file_vault_full_path = os.path.join(self.user0.get_vault_session_path(), filename)
+            self.user0.interrupt_icommand(iput_cmd, file_vault_full_path, 2048)
+            self.user0.assert_icommand('ils -L', 'STDOUT_SINGLELINE', filename)
+
+            # multiple threads could still be writing on the server side, so we need to wait for
+            # the size in the vault to converge - then were done.
```
Comment (trel) [inline scripts/irods/test/test_icommands_file_operations.py:720]: we're

File: scripts/irods/test/test_icommands_file_operations.py [hunk 2]
```diff
@@ -683,6 +683,70 @@ def test_cancel_large_iput(self):
...
+            old_size = None
+            new_size = os.path.getsize(file_vault_full_path)
+            while old_size != new_size:
+                time.sleep(1)
+                old_size = new_size
+                new_size = os.path.getsize(file_vault_full_path)
+
+            # wait for select() call to timeout, set to "SELECT_TIMEOUT_FOR_CONN", which is 60 seconds
```
Comment (trel) [inline scripts/irods/test/test_icommands_file_operations.py:728]: 'set slightly above'

**Review by trel — COMMENTED** [review]
i like it a lot.    please tweak comments and #.

File: scripts/irods/test/test_icommands_file_operations.py (hunk 1 above)
Comment (alanking) [reply scripts/irods/test/test_icommands_file_operations.py]: Done

File: scripts/irods/test/test_icommands_file_operations.py (hunk 2 above)
Comment (alanking) [reply scripts/irods/test/test_icommands_file_operations.py]: Done

**Comment by alanking** [comment]
Fixed comments, squashed, #'d

**Pull request merged**